### DISCUSSION TITLE
`useLazyFetch` calls both `onResponse` and `onResponseError`

### DISCUSSION THREAD

**Discussion by corrm** [nuxt/nuxt#29315] [Questions] [answered]
### Environment

------------------------------
- Operating System: Linux
- Node Version:     v22.9.0
- Nuxt Version:     3.13.2
- CLI Version:      3.13.2
- Nitro Version:    2.9.7
- Package Manager:  bun@1.1.24
- Builder:          -
- User Config:      compatibilityDate, devtools, srcDir, ssr, typescript, modules, eslint, i18n, security, tailwindcss, app, vite, hooks, nitro, routeRules, runtimeConfig
- Runtime Modules:  @nuxt/eslint@0.5.7, @nuxtjs/i18n@8.5.5, @nuxtjs/tailwindcss@6.12.1, nuxt-auth-utils@0.4.2, nuxt-mikro-orm-module@0.4.1, @nuxt/icon@1.5.2, @vueuse/nuxt@11.1.0
- Build Modules:    -
------------------------------

### Reproduction

```ts
    await useLazyFetch("/api/auth/login", {
        method: "POST",
        body: {
            tgInitData: initData,
        },
        onResponse: async (ctx) => {
            // if (!ctx.response.ok) {
            //     return;
            // }
            console.log(ctx);
        },
        onResponseError: (ctx) => {
            console.log("ERROR");
        },
    });
```

### Describe the bug

Both log get called:
![Image](https://github.com/user-attachments/assets/5d1eb0f5-7fa9-4369-8289-c434cc5565b0)

I can confirm(browser dev network tab) its only one http request.
if thats the expected behavior then
- What the point of `onResponseError`
- What the point of call `onResponse` and `onResponseError`.

**Comment by TheAlexLichter** [comment, marked as answer]
That's correct and intended (see https://github.com/unjs/ofetch/blob/ad320ae0ff083c7c6b094d025cf257faa3e2e5bf/src/fetch.ts#L225-L247)

You can decided to only track errors or do something on every response. Interceptors are not meant to be used for common data operations though (use try/catch, watching data or error, etc. etc.)
